### PR TITLE
Update warnings for available accelerators not being used

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Update warnings for available accelerators not being used

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1734,14 +1734,15 @@ def _log_device_info(self) -> None:
 
         if torch.cuda.is_available() and self._device_type != _AcceleratorType.GPU:
             rank_zero_warn(
-                "GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.",
+                "GPU available but not used. Set `gpus` via your trainer `Trainer(accelerator='gpu', devices=1)`"
+                " or script `--accelerator='gpu' --devices=1`.",
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1738]:
```suggestion
                "GPU available but not used. Set `accelerator` and `devices` via your trainer
                " `Trainer(accelerator='gpu', devices=1)`.",
```
should we mention about script explicitly? I think it depends upon what kind of argument parser and config they are using...

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -1734,14 +1734,15 @@ def _log_device_info(self) -> None:
...
                 category=PossibleUserWarning,
             )
 
         if _TPU_AVAILABLE and self._device_type != _AcceleratorType.TPU:
             rank_zero_warn(
-                "TPU available but not used. Set the `tpu_cores` flag in your trainer"
-                " `Trainer(tpu_cores=8)` or script `--tpu_cores=8`."
+                "TPU available but not used. Set `tpu_cores` via your trainer"
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1744]: same here.

File: pytorch_lightning/trainer/trainer.py [hunk 3]
```diff
@@ -1750,8 +1751,8 @@ def _log_device_info(self) -> None:
             and not isinstance(self.accelerator, IPUAccelerator)
         ):
             rank_zero_warn(
-                "IPU available but not used. Set the `ipus` flag in your trainer"
-                " `Trainer(ipus=8)` or script `--ipus=8`."
+                "IPU available but not used. Set `ipus` via your trainer"
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1754]: same here.

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py]: Updated!

File: pytorch_lightning/trainer/trainer.py [hunk 4]
```diff
@@ -1734,14 +1734,15 @@ def _log_device_info(self) -> None:
 
         if torch.cuda.is_available() and self._device_type != _AcceleratorType.GPU:
             rank_zero_warn(
-                "GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.",
+                "GPU available but not used. Set `accelerator` and `devices` via your"
+                " via your trainer `Trainer(accelerator='gpu', devices=1)`.",
```
Comment (carmocca) [inline pytorch_lightning/trainer/trainer.py:1738]:
```suggestion
                " trainer `Trainer(accelerator='gpu', devices=1)`.",
```

File: tests/accelerators/test_tpu.py [hunk 5]
```diff
@@ -332,3 +332,9 @@ def test_devices_auto_choice_tpu():
     trainer = Trainer(accelerator="auto", devices="auto")
     assert trainer.devices == 8
     assert trainer.tpu_cores == 8
+
+
+@RunIf(tpu=True)
+def test_warning_if_tpus_not_used(tmpdir):
+    with pytest.warns(UserWarning, match="TPU available but not used. Set `tpu_cores` via your trainer"):
+        Trainer(default_root_dir=tmpdir)
```
Comment (carmocca) [inline tests/accelerators/test_tpu.py:340]: tmpdir is only useful if we `fit`

```suggestion
def test_warning_if_tpus_not_used():
    with pytest.warns(UserWarning, match="TPU available but not used. Set `tpu_cores` via your trainer"):
        Trainer()
```

File: tests/accelerators/test_ipu.py [hunk 6]
```diff
@@ -115,7 +115,7 @@ def test_accelerator_selected(tmpdir):
 
 @RunIf(ipu=True)
 def test_warning_if_ipus_not_used(tmpdir):
```
Comment (carmocca) [inline tests/accelerators/test_ipu.py:117]: Feel free to remove tmpdir from here too

File: pytorch_lightning/trainer/trainer.py [hunk 7]
```diff
@@ -1772,14 +1772,15 @@ def _log_device_info(self) -> None:
...
+                " trainer `Trainer(accelerator='gpu', devices=1)`.",
                 category=PossibleUserWarning,
             )
 
         if _TPU_AVAILABLE and self._device_type != _AcceleratorType.TPU:
             rank_zero_warn(
-                "TPU available but not used. Set the `tpu_cores` flag in your trainer"
-                " `Trainer(tpu_cores=8)` or script `--tpu_cores=8`."
+                "TPU available but not used. Set `accelerator` and `devices` via your trainer"
+                " `Trainer(accelerator='tpu', devices=8)`."
```
Comment (ananthsub) [inline pytorch_lightning/trainer/trainer.py:1783]: This warning should provide the number of available devices, not hardcode 8

File: pytorch_lightning/trainer/trainer.py [hunk 8]
```diff
@@ -1788,8 +1789,8 @@ def _log_device_info(self) -> None:
             and not isinstance(self.accelerator, IPUAccelerator)
         ):
             rank_zero_warn(
-                "IPU available but not used. Set the `ipus` flag in your trainer"
-                " `Trainer(ipus=8)` or script `--ipus=8`."
+                "IPU available but not used. Set `accelerator` and `devices` via your trainer"
+                " `Trainer(accelerator='ipu', devices=8)`."
```
Comment (ananthsub) [inline pytorch_lightning/trainer/trainer.py:1793]: Same here

File: pytorch_lightning/trainer/trainer.py (hunk 7 above)
Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py:1783]: TPUs always come up with the set of 8 devices. But yes, we could update this for ipus, gpus, etc.

Also, when the user does `Trainer(accelerator="tpu")`, the devices are set to `auto_device_count`. That's the default.

Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py]: Updated!

File: pytorch_lightning/trainer/trainer.py [hunk 9]
```diff
@@ -1772,14 +1772,15 @@ def _log_device_info(self) -> None:
 
         if torch.cuda.is_available() and self._device_type != _AcceleratorType.GPU:
             rank_zero_warn(
-                "GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.",
+                "GPU available but not used. Set `accelerator` and `devices` via your"
+                f" trainer `Trainer(accelerator='gpu', devices={GPUAccelerator.auto_device_count()})`.",
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1776]:
```suggestion
                "GPU available but not used. Set `accelerator` and `devices` using"
                f" `Trainer(accelerator='gpu', devices={GPUAccelerator.auto_device_count()})`.",
```

File: pytorch_lightning/trainer/trainer.py [hunk 10]
```diff
@@ -1772,14 +1772,15 @@ def _log_device_info(self) -> None:
...
                 category=PossibleUserWarning,
             )
 
         if _TPU_AVAILABLE and self._device_type != _AcceleratorType.TPU:
             rank_zero_warn(
-                "TPU available but not used. Set the `tpu_cores` flag in your trainer"
-                " `Trainer(tpu_cores=8)` or script `--tpu_cores=8`."
+                "TPU available but not used. Set `accelerator` and `devices` via your trainer"
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1782]:
```suggestion
                "TPU available but not used. Set `accelerator` and `devices` using"
```

File: pytorch_lightning/trainer/trainer.py [hunk 11]
```diff
@@ -1788,8 +1789,8 @@ def _log_device_info(self) -> None:
             and not isinstance(self.accelerator, IPUAccelerator)
         ):
             rank_zero_warn(
-                "IPU available but not used. Set the `ipus` flag in your trainer"
-                " `Trainer(ipus=8)` or script `--ipus=8`."
+                "IPU available but not used. Set `accelerator` and `devices` via your trainer"
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/trainer.py:1792]:
```suggestion
                "IPU available but not used. Set `accelerator` and `devices` using"
```

**Review by rohitgr7 — APPROVED** [review]
nit:

**Review by awaelchli — APPROVED** [review]
